client/src/App.js

```diff
@@ -85,7 +86,7 @@ class App extends Component {
   render() {
     return (
       <div className="App">
-        <Select store={store} />
+        <Select store={store} process={action => agent.process(action)} />
```

Comment:
The `agent` is in scope here, so we wrap its process method and pass it down. React Context could also be used, if desired.